src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -123,6 +124,65 @@ namespace xclhwemhal2 {
     return buf_size;
   }
 
+  void HwEmShim::writeNewSimulateScript(std::string simPath, std::string simName, std::string simCmd, std::string guiOrBatch, std::string guiEnableOption)
+  {
+
+    std::stringstream ss;
+    ss << simPath << "/simulate.sh";
+    std::string filePth = ss.str();
+
+    std::ifstream seek1;
+    std::ofstream temp1;
+    seek1.open(filePth);
+    temp1.open(simPath + "/temp.sh");
+    std::string temp;
+
+    while (getline(seek1, temp))
+    {
+      if (temp.find(simCmd) != std::string::npos)
+      {
+        if (boost::iequals(simName, "xcelium")) {
+          //for xcelium, append -gui to the command in simulate.sh to run in GUI mode if it doesnt exist
+          if (temp.find(guiOrBatch) == std::string::npos) {
+            temp.append(guiEnableOption);
+          }
+        }
+        else if (boost::iequals(simName, "questa")) {
+          //for questa, replace -c with  -gui to the command in simulate.sh to run in GUI mode
+          if (temp.find(guiOrBatch) != std::string::npos) {
+            temp = std::regex_replace(temp, std::regex(guiOrBatch), guiEnableOption);
+          }
+        }
+
+      }
+      temp1 << temp << std::endl;
+    }
+
+    std::remove((simPath + "/simulate.sh").c_str());
```

Comment:
Is this an obfuscation contest here? ;-)

Reply:
It took me some time to understand that you are using https://en.cppreference.com/w/cpp/io/c/remove and not https://en.cppreference.com/w/cpp/filesystem/rename